src/App.tsx

```diff
@@ -1,14 +1,104 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './enums/Status';
+import { User } from './types/User';
+import { SetTodo } from './interfaces/interfaces';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(true);
+  const [loadingUser, setLoadingUser] = useState(true);
+  const [text, setText] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [usersId, setUsersId] = useState<number>(0);
+  const [user, setUser] = useState<User | null>(null);
+  const [choseTodo, setChoseTodo] = useState<SetTodo>({
+    id: null,
+    title: '',
+    completed: false,
+    highlighteTodo: false,
+  });
```

Comment:
```suggestion
  const [choseTodo, setChoseTodo] = useState<SetTodo>(null);
```
set null or todo in choseTodo state. after that you don't need showModal state. will be enough show or not modal depends on chose todo. less amount of states can help to solve problem with todo modal.